code/src/sixsq/nuvla/server/resources/spec/voucher.cljc

```diff
@@ -0,0 +1,189 @@
+(ns sixsq.nuvla.server.resources.spec.voucher
+  (:require
+    [clojure.spec.alpha :as s]
+    [sixsq.nuvla.server.resources.spec.common :as cimi-common]
+    [sixsq.nuvla.server.resources.spec.core :as cimi-core]
+    [sixsq.nuvla.server.util.spec :as su]
+    [spec-tools.core :as st]))
+
+
+(s/def ::owner ::cimi-common/resource-link)
+(s/def ::user ::cimi-common/resource-link)
+
+(s/def ::amount
+  (-> (st/spec double)
+      (assoc :name "amount"
+             :json-schema/type "double"
+             :json-schema/editable true
+
+             :json-schema/display-name "voucher amount"
+             :json-schema/description "amount of the voucher"
+             :json-schema/group "body"
+             :json-schema/order 20
+             :json-schema/hidden false
+             :json-schema/sensitive false)))
+
+
+(s/def ::currency
+  (-> (st/spec ::cimi-core/nonblank-string)
+      (assoc :name "currency"
+             :json-schema/type "string"
+             :json-schema/editable true
+
+             :json-schema/display-name "currency"
+             :json-schema/description "currency for the voucher amount"
+             :json-schema/group "body"
+             :json-schema/order 21
+             :json-schema/hidden false
+             :json-schema/sensitive false)))
+
+
+(s/def ::expiration-date
+  (-> (st/spec ::cimi-core/timestamp)
+      (assoc :name "expiration-date"
+             :json-schema/type "date-time"
+             :json-schema/editable true
+
+             :json-schema/display-name "expiration-date"
+             :json-schema/description "expiration-date for the voucher"
+             :json-schema/group "body"
+             :json-schema/order 22
+             :json-schema/hidden false
+             :json-schema/sensitive false)))
+
+
+(s/def ::state
+  (-> (st/spec #{"new", "activated",
+                 "expired", "redeemed"})
+      (assoc :name "state"
+             :json-schema/type "string"
+             :json-schema/editable true
+
+             :json-schema/display-name "state"
+             :json-schema/description "usage state of the voucher"
+             :json-schema/group "body"
+             :json-schema/order 23
+             :json-schema/hidden false
+             :json-schema/sensitive false
+
+             :json-schema/value-scope {:values  ["new", "activated",
+                                                 "expired", "redeemed"]
+                                       :default "new"})))
+
+
+(s/def ::service-info
```

Comment:
Shouldn't this point to a `infrastructure-service` or `infrastructure-service-resource`?  In the ISG there's a link for documentation.

Reply:
no. maybe this naming is not the best, but this is purely a url of some sort, to the service provider landing page or voucher registration page

Reply:
this name was not my first choice, but it was asked by ocre and in lack of a better one, that's what I've defined. If you find a better name let me know

Reply:
renamed to `service-info-url`